### PR TITLE
CI: Uploads build APKs as artifacts at GH-Actions

### LINKED ISSUE #2484: CI: Upload Build APKs as Artifacts at GH-Actions
**Is your feature request related to a problem? Please describe.**
As of now, to test the app on various online emulators, we need to build APK of the concerned module locally from AS and then use it. Doing so takes time, so it'll be better if we associate APK build and uploading as artifacts to GH Actions with each merge-commit and/or Pull Request.

**Describe the solution you'd like**
A new job in the concerned workflow file for uploading the concerned artifact.

**Describe alternatives you've considered**
Currently, the proposed solution seems unique to me.

**Additional context**
We need to ponder upon whether we need the APKs associated with each merge commit or normal commit or not. But I'm pretty sure with the need of this functionality with the PRs.

### PR DESCRIPTION
**Pull request by s-ayush2903**
Close #2484 

Wrote a job to upload APK on GH actions and to the usual apk path in the project.

**Screenshots** 
NA

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -27,3 +27,9 @@ jobs:
 
       - name: Build all configurations
         run: ./gradlew assemble
+
+      - name: Upload APK as Artifacts
+        uses: actions/upload-artifact@v2.2.0
+        with:
+          name: kiwix-android
+          path: app/build/outputs/apk/debug
```
Comment (macgills) [inline .github/workflows/pull_request.yml:35]: you can use wildcards no? Might have to check the upload-artifact docs. `**.apk`, that'll keep this future proof and ensure we get the custom app in here too

Comment (s-ayush2903) [reply .github/workflows/pull_request.yml]: Nicee, I didn't know about wildcards what they are and when to use 'em. Got it now, thanks!

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -27,3 +27,11 @@ jobs:
 
       - name: Build all configurations
         run: ./gradlew assemble
+
+      - name: Upload APK as Artifacts
+        uses: actions/upload-artifact@v2.2.0
+        with:
+          name: kiwix-android
+          path: |
+            app/**/*.apk
```
Comment (macgills) [inline .github/workflows/pull_request.yml:36]: I am not sure what you tried but I think these 2 lines are replaceable with `**/**.apk`

Comment (macgills) [reply .github/workflows/pull_request.yml:36]: or `**/*.apk` it is always tough to say what regex syntax somebody is using

Comment (s-ayush2903) [reply .github/workflows/pull_request.yml]: Yep, I used it but it was a wrong way I remember it, I used it like:
`path: **/*.apk`

It then didn't even trigger the CI, so I removed it.

**Review by macgills — APPROVED** [review]

**Pull request merged**